index.bs

```diff
@@ -221,6 +221,25 @@ and doing cleanup as necessary (see the [[#listen-fully-active]] pattern above).
   but every feature specification should think about this and decide what works best for the feature.
 </div>
 
+<h4 id="non-synchronous-steps">Be aware that navigations can happen between steps running [=in parallel=]</h4>
+While running steps [=in parallel=],
+the document the steps are running in or are otherwise using might be navigated away and (later) navigated to again.
```

Comment:
This is still confusing. The document represents the conceptual "main thread". "In parallel" is a concept representing another thread/process. So if the steps run in parallel, they are by definition not running on the main thread.

Task queued steps might, of course, but then you need to spell that out. As you do below.

Reply:
OK, how about:
"While running steps [=in parallel=], if the steps are in a queued task associated with the document,
the document might be navigated away and (later) on navigated to again. The steps themselves might not run while the document is not [=fully active=], and a considerable amount of time might have passed between the steps,
so the user might not expect some states that are saved in the steps to be preserved. "

Reply:
Still unclear to me. Is this what you mean:

> If steps that run in parallel queue a task that is associated with a document, you need to be aware that the document might have been navigated away from and (later on) navigated to again. This can happen while running the steps in parallel and can have happened when the steps in the queued task eventually run. ...

?

Reply:
Thanks! I used that now, lmk if there are things that still need editing :)